test/utils/utils.go

```diff
@@ -90,54 +90,65 @@ type HostOS struct {
 }
 
 // MakeOptions assembles all podman options
-func (p *PodmanTest) MakeOptions(args []string, noEvents, noCache bool) []string {
-	return p.PodmanMakeOptions(args, noEvents, noCache)
+func (p *PodmanTest) MakeOptions(args []string, options PodmanExecOptions) []string {
+	return p.PodmanMakeOptions(args, options)
 }
 
-// PodmanAsUserBase exec podman as user. uid and gid is set for credentials usage. env is used
-// to record the env for debugging
-func (p *PodmanTest) PodmanAsUserBase(args []string, uid, gid uint32, cwd string, env []string, noEvents, noCache bool, wrapper []string, extraFiles []*os.File) *PodmanSession {
+// PodmanExecOptions modify behavior of PodmanTest.PodmanExecBaseWithOptions and its callers.
+// Users should typically leave most fields default-initialized, and only set those that are relevant to them.
+type PodmanExecOptions struct {
+	UID, GID   uint32   // default: inherited form the current process
+	CWD        string   // default: inherited form the current process
+	Env        []string // default: inherited form the current process
+	NoEvents   bool
+	NoCache    bool
+	Wrapper    []string // A command to run, receiving the Podman command line. default: none
+	ExtraFiles []*os.File
+}
+
+// PodmanExecBaseWithOptions execs podman with the specified args, and in an environment defined by options
+func (p *PodmanTest) PodmanExecBaseWithOptions(args []string, options PodmanExecOptions) *PodmanSession {
```

Comment:
This name is fairly ridiculous … I considered reusing the now-available `PodmanBase` name, but I left it as is because it _seems_ to me that most users should call something higher-level, and a long tedious name is a _bit_ of a hint to use something else.

Reply:
it is a test suite, fine by me.